tests/language/json.liq

```diff
@@ -208,25 +199,45 @@ def f() =
 
   data = '{"aabbcc": 34, "ddeerr": 54 }'
   let json.parse (x : [(string * int)] as json.object) = data
-  t(list.assoc("aabbcc", x), 34)
-  t(list.assoc("ddeerr", x), 54)
+  test.equals(list.assoc("aabbcc", x), 34)
+  test.equals(list.assoc("ddeerr", x), 54)
 
   data = '{ "foo": 123 }'
   let json.parse ( x : {
     foo : string
   }?) = data
-  t(x, null())
+  test.equals(x, null())
 
   data = '[ "gni", 123 ]'
   let json.parse ( x : [int]? ) = data
-  t(x, null())
+  test.equals(x, null())
 
   let json.parse ( x : (string * int * bool)? ) = data
-  t(x, null())
+  test.equals(x, null())
 
   data = '[ "gni", 123, "gno" ]'
   let json.parse ( x : (string * int) ) = data
-  t(x, ("gni",123))
+  test.equals(x, ("gni",123))
+
+  data = '{
+    "foo": {
+      "gni": {
+        "bla": 123
+      }
+    }
+  }'
+  let json.parse x = data
+  test.equals(x?.foo.gni?.bla, null(123))
```

Comment:
Do we really need the cast here?
```suggestion
  test.equals(x?.foo.gni?.bla, 123)
```

Reply:
Not for the value test but for the type test I think it's good to have, to make sure that this returning `int?`.